website/addons/base/serializer.py

```diff
@@ -165,3 +165,50 @@ def serialize_citation(self, citation):
             'kind': 'file',
             'id': citation['id'],
         }
+
+
+# TODO: change name?
+class GenericAddonSerializer(OAuthAddonSerializer):
```

Comment:
All of the methods here that match exactly with CitationsAddonSerializer should be moved to OAuthAddonSerializer and deleted from both subclasses (serialized_urls, user_is_owner, credentials_owner).

Then you can remove GenericAddonSerializer, and move your implementation of serialized_node_settings to GoogleDriveSerializer (which will now subclass OAuthAddonSerializer).